zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/StateControllerImpl.java

```diff
@@ -279,6 +284,20 @@ private void takeSnapshot(
         });
   }
 
+  @Override
+  public ActorFuture<PersistedSnapshot> takeSnapshot(
+      final int partition, final long lastProcessedPosition) {
+    if (partition != partitionId) {
```

Comment:
:wrench: Looks like we could change this method in the interface to just not take a partition id anymore

Reply:
Totally agree, it's simpler to just assume that each instance it's already scoped to one partition. I already had in mind to change once everything is in place :+1: